### ISSUE TITLE
MA0002 raised on first line of multi-line LINQ chain

### ISSUE THREAD

**Issue by YoshiRulz** [meziantou/Meziantou.Analyzer#789]
### Version of the Meziantou.Analyzer NuGet package

2.0.163

### Rule Identifier

MA0002

### Target Framework

netstandard2.0, net48

### C# Language version

C# 12

### Description

When you have a LINQ chain with `Contains`, `Distinct`, etc., the location of all "Use an overload that has [...an] IComparer<string> parameter" diagnostics is the first line of the chain rather than where the offending calls are.

### Reproduction Steps

Minimal code:

```c#
string[] strings = [ "Hel", "lo,", " wo", "rld" ];
Console.WriteLine(strings // both reported here
	.Distinct()
	.Skip(1)
	.Contains("rld"));
```

**Comment by meziantou** [comment]
Currently, it reports the `InvocationExpression` having an issue. In your case, there are multiple nested `InvocationExpression`, all starting at the same location. So, the locations of both diagnostics overlap and it's not clear which method is problematic for the first diagnostic. 

I'm not sure how _standard_ it would be to set the location on the method name + arguments instead of the full expression. It's possible, I'm just not sure what other analyzers do. Also, should it be only for this rule or all rules reporting an `InvocationExpression`?

Note that displaying diagnostics inline make it easier to know where the issue is:

![Image](https://github.com/user-attachments/assets/3173bc07-f1c8-4797-8528-04f92369e2da)

**Comment by YoshiRulz** [comment]
Oh, because the receiver for each call is the whole chain up to that point, that makes sense. I'm guessing VS is using the span end to know where to put those chips?

As a point of comparison, CA1826 and CA1829 seem to behave this way too. I prefer how you've done it for MA0020 though.

**Comment by meziantou** [comment]
We can give it a try and see if the UX is better
